packages/fetch-engine/src/download.ts

```diff
@@ -294,37 +291,44 @@ async function binaryNeedsToBeDownloaded(
   // If there is no cache and the file doesn't exist, we for sure need to download it
   if (!targetExists) {
     debug(`file ${job.targetFilePath} does not exist and must be downloaded`)
+
     return true
   }
 
   // 3. If same platform, always check --version
-  if (job.binaryTarget === nativePlatform && job.binaryName !== BinaryType.libqueryEngine) {
-    const works = await checkVersionCommand(job.targetFilePath)
-    return !works
-  } // TODO: this is probably not useful anymore
+  if (job.binaryTarget === nativePlatform) {
+    const currentVersion = await getVersion(job.targetFilePath, job.binaryName)
 
-  return false
-}
+    if (currentVersion?.includes(version) === false) {
```

Comment:
We check that the current binary/file/engine is aligned with the version we want.